cineast-api/src/main/java/org/vitrivr/cineast/api/APIEndpoint.java

```diff
@@ -122,66 +166,175 @@ private static Service dispatchService(boolean secure) {
+                    if(route.isResponseCollection()){
+                        builder.withResponseAsCollection(route.outClass());
+                    }else{
+                        builder.withResponseType(route.outClass());
+                    }
+                    builder.withRequestType(route.inClass()).withDescription(route.descriptionForPost());
+                    apiEndpoint.post(builder, route);
+                    break;
+                case put:
+                    builder = path(makePath(route.routeForPost()));
+                    if(route.isResponseCollection()){
+                        builder.withResponseAsCollection(route.outClass());
+                    }else{
+                        builder.withResponseType(route.outClass());
+                    }
+                    builder.withRequestType(route.inClass()).withDescription(route.descriptionForPost());
+                    apiEndpoint.put(builder, route);
+                    break;
+                case delete:
+                    builder = path(makePath(route.routeForPost()));
+                    if(route.isResponseCollection()){
+                        builder.withResponseAsCollection(route.outClass());
+                    }else{
+                        builder.withResponseType(route.outClass());
+                    }
+                    builder.withRequestType(route.inClass()).withDescription(route.descriptionForPost());
+                    apiEndpoint.delete(builder, route);
+                    break;
+                default:
+                    LOGGER.warn("Route {} supports unsupported http method {}",route, method);
+            }
+        });
+    }
+
+    @SuppressWarnings("unchecked")
+    private void registerRoute(final DocumentedRestOperation route, final Service service){
+        route.supportedMethods().forEach(m -> {
+            HttpMethod method = (HttpMethod)m;
+            MethodDescriptor.Builder builder;
+            switch(method){
+                case get:
+                    service.get(makePath(route.routeForGet()), route);
+                    break;
+                case post:
+                    service.post(makePath(route.routeForPost()), route);
+                    break;
+                case put:
+                    service.put(makePath(route.routeForPut()), route);
+                    break;
+                case delete:
+                    service.delete(makePath(route.routeForDelete()), route);
+                    break;
+                default:
+                    LOGGER.warn("Route {} supports unsupported http method {}",route, method);
+            }
+        });
+    }
+
+    @Override
+    public void bind(final SparkSwagger restApi){
+        ApiEndpoint apiEndpoint = registerSparkSwaggerEndpoint(restApi);
+        registerRoutesOnEndpoint(apiEndpoint);
+    }
+
+    private void registerOperations(){
+        /* Add your operations here */
+        registeredOperations.addAll(
```

Comment:
Why does a private method populate an externally defined data structure? The list should be created and returned here in order to ensure idempotence.